### DISCUSSION TITLE
"Nginx -s reload, exit status 1" on all SSL requests

### DISCUSSION THREAD

**Discussion by Decoherent** [nginx-proxy/nginx-proxy#1650] [Q&A]
I'm having some trouble getting nginx-proxy to work nicely on my machine with FreshRSS. The two work together fine on port 80, but when I'm trying to use SSL, I'm getting the following error, which is thrown as soon as the other container is started:

`Error running notify command: nginx -s reload, exit status 1`

I'm running these from the command line via docker run, so that I can see what's going on! Here's my configuration:
(previously created the bridge network "freshrss-network", as suggested by their documentation)

```
docker run \
  -p 80:80 \
  -p 443:443 \
  --net freshrss-network \
  -v /etc/certs:/etc/nginx/certs \
  -v /var/run/docker.sock:/tmp/docker.sock:ro \
  --name nginx-proxy jwilder/nginx-proxy:alpine
  
docker run --restart unless-stopped --log-opt max-size=10m \
  -v freshrss-data:/var/www/FreshRSS/data \
  -v freshrss-extensions:/var/www/FreshRSS/extensions \
  -e 'CRON_MIN=*/15' \
  --expose 80 \
  --expose 443 \
  --net freshrss-network \
  -e 'VIRTUAL_HOST=[my.hostname]' \
  --name freshrss freshrss/freshrss:latest

ls -l /etc/certs
total 12
-rw------- 1 root root  802 Jun  5 23:02 [my.hostname].crt
-rw------- 1 root root  242 Jun  5 23:02 [my.hostname].key
-rw------- 1 root root 1144 Jun  6 00:16 [my.hostname].pem
```
Does anyone have any suggestions? This is simple enough I feel like I'm banging my head against something trivial.

edit: Sorry, I should have posted the whole error log:
```
WARNING: /etc/nginx/dhparam/dhparam.pem was not found. A pre-generated dhparam.pem will be used for now while a new one
is being generated in the background.  Once the new dhparam.pem is in place, nginx will be reloaded.
Generating DSA parameters, 4096 bit long prime
forego     | starting dockergen.1 on port 5000
forego     | starting nginx.1 on port 5100
dockergen.1 | 2021/06/06 06:01:40 Generated '/etc/nginx/conf.d/default.conf' from 1 containers
dockergen.1 | 2021/06/06 06:01:40 Running 'nginx -s reload'
dockergen.1 | 2021/06/06 06:01:40 Watching docker events
dockergen.1 | 2021/06/06 06:01:41 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'
2021/06/06 06:01:42 [notice] 46#46: signal process started
dhparam generation complete, reloading nginx
dockergen.1 | 2021/06/06 06:02:27 Received event start for container 7ee96dfaeae1
dockergen.1 | 2021/06/06 06:02:27 Generated '/etc/nginx/conf.d/default.conf' from 2 containers
dockergen.1 | 2021/06/06 06:02:27 Error running notify command: nginx -s reload, exit status 1
```

Also, when trying to connect to port 80 with this setup, I get a 503 error, but there's an nginx event in the log. When I try to connect via https, there's nothing.

**Comment by Glydric** [comment]
I had the same problem, but then notice an error on defining env (VIRTUAL_POST instead of VIRTUAL_PORT) also i suggest you to use docker compose

**Comment by tkw1536** [comment]
For `freshrss` you shouldn't have the container itself serve `https` as this is handled by `nginx-proxy`.
`nginx-proxy` will terminate the HTTPS for you, and redirect it to the http port of the `freshrss` container.
You also shouldn't `--expose` port 80 and 443, instead you should set `VIRTUAL_PORT=80` on the container like so:

```
docker run --restart unless-stopped --log-opt max-size=10m \
  -v freshrss-data:/var/www/FreshRSS/data \
  -v freshrss-extensions:/var/www/FreshRSS/extensions \
  -e 'CRON_MIN=*/15' \
  --net freshrss-network \
  -e 'VIRTUAL_HOST=[my.hostname]' \
  -e 'VIRTUAL_PORT=80' \
  --name freshrss freshrss/freshrss:latest
```

I also second @Glydric22, please use `docker compose` if you can. 
It makes configurations with multiple containers a lot easier to handle.
